stackit/internal/services/serviceaccount/account/resource.go

```diff
@@ -0,0 +1,363 @@
+}
+
+// Delete deletes the service account and removes it from the Terraform state on success.
+func (r *serviceAccountResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) { // nolint:gocritic // function signature required by Terraform
+	// Retrieve current state of the resource.
+	var model Model
+	diags := req.State.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectId := model.ProjectId.ValueString()
+	serviceAccountName := model.Name.ValueString()
+	serviceAccountEmail := model.Email.ValueString()
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "service_account_name", serviceAccountName)
+
+	// Call API to delete the existing service account.
+	err := r.client.DeleteServiceAccount(ctx, projectId, serviceAccountEmail).Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error deleting service account", fmt.Sprintf("Calling API: %v", err))
+		return
+	}
+	tflog.Info(ctx, "Service account deleted")
+}
+
+// ImportState imports a resource into the Terraform state on success.
+// The expected format of the resource import identifier is: project_id,email
+func (r *serviceAccountResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Split the import identifier to extract project ID and email.
+	idParts := strings.Split(req.ID, core.Separator)
+
+	// Ensure the import identifier format is correct.
+	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+		core.LogAndAddError(ctx, &resp.Diagnostics,
+			"Error importing service account",
+			fmt.Sprintf("Expected import identifier with format: [project_id],[email]  Got: %q", req.ID),
+		)
+		return
+	}
+
+	projectId := idParts[0]
+	email := idParts[1]
+
+	// Attempt to parse the name from the email if valid.
+	name, err := parseNameFromEmail(email)
+	if name != "" && err == nil {
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("name"), name)...)
+	}
+
+	// Set the project ID and email attributes in the state.
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), projectId)...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("email"), email)...)
+	tflog.Info(ctx, "Service account state imported")
+}
+
+// toCreatePayload generates the payload to create a new service account.
+func toCreatePayload(model *Model) (*serviceaccount.CreateServiceAccountPayload, error) {
+	if model == nil {
+		return nil, fmt.Errorf("nil model")
+	}
+
+	return &serviceaccount.CreateServiceAccountPayload{
+		Name: conversion.StringValueToPointer(model.Name),
+	}, nil
+}
+
+// mapCreateOrListResponse maps a ServiceAccount response to the model.
+func mapCreateOrListResponse(resp *serviceaccount.ServiceAccount, model *Model) error {
```

Comment:
As there is no other map function needed, I would suggest to rename it to `mapFields` like in other resources
```suggestion
func mapFields(resp *serviceaccount.ServiceAccount, model *Model) error {
```